src/mscorlib/shared/System/Number.NumberBuffer.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class Number
+    {
+        // Constants used by number parsing
+        private const Int32 NumberMaxDigits = 32; // needs to == NUMBER_MAXDIGITS in coreclr's src/classlibnative/bcltype/number.h
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct NumberBuffer // needs to match layout of NUMBER in coreclr's src/classlibnative/bcltype/number.h
```

Comment:
>needs to match layout of NUMBER 

It's not quite matching right now - the C++ version has an "alldigits" field - only used for BigInt but if you're passing these down via FCalls, it's playing with fire not to have the space here.

Reply:
Yup, thanks for the catch. Will fix.